File: WebClient SDK/src/default/workplaces/manager/managerWorkplaceExtension.ts
import { ManagerWorkplaceViewModel, ManagerWorkplace } from './managerWorkplace';
import { ManagerWorkplaceSettings } from './managerWorkplaceSettings';
import { ApplicationExtension } from 'tessa';
import { WorkplaceViewComponentExtension } from 'tessa/ui/views/extensions';
import { IWorkplaceViewComponent, ViewComponentRegistry, StandardViewComponentContentItemFactory } from 'tessa/ui/views';

//#region ManagerWorkplaceExtension

export class ManagerWorkplaceExtension extends WorkplaceViewComponentExtension {

  public getExtensionName(): string {
    return 'Tessa.Extensions.Default.Client.Workplaces.Manager.ManagerWorkplaceExtension';
  }

  public initialize(model: IWorkplaceViewComponent) {
    const settings = new ManagerWorkplaceSettings(this.settingsStorage);
    if (!settings.cardId) {
      console.error('ManagerWorkplaceExtension settings is not valid.')
      return;
    }

    model.contentFactories.set(StandardViewComponentContentItemFactory.Table,
      c => new ManagerWorkplaceViewModel(settings, c)
    );
  }

}

//#endregion

//#region ManagerWorkplaceExtension

export class ManagerWorkplaceInitializeExtension extends ApplicationExtension {

  public initialize() {
    ViewComponentRegistry.instance.register(ManagerWorkplaceViewModel, () => ManagerWorkplace);
  }

}

//#endregion

File: Source/Extensions/Tessa.Extensions.Default.Client/Workplaces/Manager/ManagerWorkplaceExtension.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="ManagerWorkplaceExtension.cs" company="Syntellect">
//   Tessa Project
// </copyright>
// --------------------------------------------------------------------------------------------------------------------

using System.Collections.Generic;

namespace Tessa.Extensions.Default.Client.Workplaces.Manager
{
    #region

    using System;

    using Tessa.Properties.Resharper;
    using Tessa.UI.Views;
    using Tessa.UI.Views.Extensions;
    using Unity;
    using Tessa.Views;
    using Tessa.Extensions.Default.Shared.Workplaces;

    #endregion

    /// <summary>
    ///     Расширение рабочего места руководителя
    /// </summary>
    /// <remarks>
    /// У расширения есть конфигуратор <see cref="ManagerWorkplaceExtensionConfigurator"/>.
    /// </remarks>
    public sealed class ManagerWorkplaceExtension : IWorkplaceViewComponentExtension, IWorkplaceExtensionSettingsRestore
    {
        /// <summary>
        /// The container.
        /// </summary>
        [NotNull]
        private readonly IUnityContainer container;

        [NotNull]
        private readonly ImageCache imageCache;

        /// <summary>
        /// The settings.
        /// </summary>
        [NotNull]
        private ManagerWorkplaceSettings settings;

        /// <inheritdoc />
        public ManagerWorkplaceExtension([NotNull] IUnityContainer container, [NotNull] ImageCache imageCache)
        {
            if (container == null)
            {
                throw new ArgumentNullException("container");
            }

            if (imageCache == null)
            {
                throw new ArgumentNullException("imageCache");
            }

            this.container = container;
            this.imageCache = imageCache;
        }

        /// <inheritdoc />
        public void Clone(IWorkplaceViewComponent source, IWorkplaceViewComponent cloned, ICloneableContext context)
        {
        }

        /// <inheritdoc />
        public void Initialize(IWorkplaceViewComponent model)
        {
            model.ContentFactories[StandardViewComponentContentItemFactory.Table] =
                c => new ManagerWorkplaceContentItem(c, this.imageCache, this.settings);
        }

        /// <inheritdoc />
        public void Initialized(IWorkplaceViewComponent model)
        {
        }

        /// <summary>
        /// Осуществляет восстановление настроек расширения
        /// </summary>
        /// <param name="metadata">
        /// Сериализованные метаданные настроек
        /// </param>
        public void Restore(Dictionary<string, object> metadata) =>
            this.settings = (ManagerWorkplaceSettings) ExtensionSettingsSerializationHelper.DeserializeDictionary<ManagerWorkplaceSettings>(metadata);
    }
}